src/main/scala/org/broadinstitute/dsde/firecloud/service/StatusService.scala

```diff
@@ -1,43 +1,62 @@
+import spray.http.StatusCodes
+import spray.httpx.SprayJsonSupport
+import spray.json.DefaultJsonProtocol._
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.ExecutionContext
+
+/**
+ * Created by anichols on 4/5/17.
+ */
+object StatusService {
+  def props(statusServiceConstructor: () => StatusService): Props = Props(statusServiceConstructor())
+
+  def constructor(app: Application)()(implicit executionContext: ExecutionContext): StatusService = {
+    new StatusService(app.searchDAO, app.agoraDAO, app.thurloeDAO, app.rawlsDAO)
+  }
+
+  case class CollectStatusInfo()
+}
 
-import spray.routing._
+class StatusService (val searchDAO: SearchDAO,
+                     val agoraDAO: AgoraDAO,
+                     val thurloeDAO: ThurloeDAO,
+                     val rawlsDAO: RawlsDAO)
+                    (implicit protected val executionContext: ExecutionContext) extends Actor with SprayJsonSupport {
 
+  def collectStatusInfo(): Future[PerRequestMessage] = {
+    val subsystemExceptionHandler: PartialFunction[Throwable, SubsystemStatus] = {
+      case e: Exception => SubsystemStatus(false, Some(Array(e.getMessage)))
+    }
 
-import org.broadinstitute.dsde.firecloud.FireCloudConfig
+    for {
+      rawlsStatus <- rawlsDAO.status recover subsystemExceptionHandler
+      thurloeStatus <- thurloeDAO.status recover subsystemExceptionHandler
+      agoraStatus <- agoraDAO.status recover subsystemExceptionHandler
+      searchStatus <- searchDAO.status recover subsystemExceptionHandler
+    } yield {
+      val statusMap = Map("Rawls" -> rawlsStatus, "Thurloe" -> thurloeStatus, "Agora" -> agoraStatus, "Search" -> searchStatus)
 
-class StatusServiceActor extends Actor with StatusService {
-  def actorRefFactory = context
-  def receive = runRoute(routes)
-}
+      if (statusMap.values.count(_.ok == false) == 0)
+        RequestComplete(SystemStatus(true, statusMap))
+      else
+        RequestComplete(StatusCodes.InternalServerError, SystemStatus(false, statusMap))
 
-trait StatusService extends HttpService with PerRequestCreator with FireCloudDirectives {
-
-  private final val ApiPrefix = "status"
-  private final val dateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSZ")
-  private implicit val executionContext = actorRefFactory.dispatcher
-
-  val routes: Route =
-    pathPrefix(ApiPrefix) {
-      pathEnd {
-        respondWithJSON {
-          complete {
-            JsObject(
-              "workspacesUrl" -> JsString(FireCloudConfig.Rawls.baseUrl + "/workspaces"),
-              "methodsUrl" -> JsString(MethodsService.remoteMethodsUrl),
-              "timestamp" -> JsString(dateFormat.format(new Date()))
-            ).toString
-          }
-        }
-      } ~ path("ping") {
-        complete {
-          dateFormat.format(new Date())
-        }
-      }
     }
-}
+  }
+
+  def receive = {
+    case CollectStatusInfo => collectStatusInfo() pipeTo sender
+  }
```

Comment:
by convention, we put the `receive` method at the top of the class, and specify `override`.